### ISSUE TITLE
Appsignal Logger backend crashes when running tests

### ISSUE THREAD

**Issue by wkirschbaum** [appsignal/appsignal-elixir#836]
When adding the logger to Application.start/2: 

```elixir
Logger.add_backend({Appsignal.Logger.Backend, group: "phoenix"})
```

the logging works as expected, but when running automated tests with `mix test` the logger very often crashes with the following stacktrace: 

```
 [error] GenServer #PID<0.2511.0> terminating
** (stop) {:EXIT, {:function_clause, [{Appsignal.Logger.Backend, :handle_event, [:flush, [format: :plaintext, group: "phoenix"]], [file: 'lib/appsignal/logger/backend.ex', line: 8]}, {:gen_event, :server_update, 4, [file: 'gen_event.erl', line: 802]}, {:gen_event, :server_notify, 4, [file: 'gen_event.erl', line: 784]}, {:gen_event, :handle_msg, 6, [file: 'gen_event.erl', line: 529]}, {:proc_lib, :init_p_do_apply, 3, [file: 'proc_lib.erl', line: 240]}]}}
Last message: {:gen_event_EXIT, {Appsignal.Logger.Backend, [group: "phoenix"]}, {:EXIT, {:function_clause, [{Appsignal.Logger.Backend, :handle_event, [:flush, [format: :plaintext, group: "phoenix"]], [file: 'lib/appsignal/logger/backend.ex', line: 8]}, {:gen_event, :server_update, 4, [file: 'gen_event.erl', line: 802]}, {:gen_event, :server_notify, 4, [file: 'gen_event.erl', line: 784]}, {:gen_event, :handle_msg, 6, [file: 'gen_event.erl', line: 529]}, {:proc_lib, :init_p_do_apply, 3, [file: 'proc_lib.erl', line: 240]}]}}}
```

The above log is being generated by the console logger.

**Comment by jeffkreeftmeijer** [comment]
Hey @wkirschbaum,

Thanks for reporting this. We’re reviewing a patch for this in https://github.com/appsignal/appsignal-elixir/pull/837. If you have time, could you try running your test suite after switching to the `logger-backend-error` branch?

**Comment by jeffkreeftmeijer** [comment]
Fixed in 2.6.1 (#837)
